lib/cli/diskpart.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const os = require('os')
+const fs = require('fs')
+const path = require('path')
+const crypto = require('crypto')
+const childProcess = require('child_process')
+const debug = require('debug')('etcher:cli:diskpart')
+const Promise = require('bluebird')
+const retry = require('bluebird-retry')
+
+const TMP_RANDOM_BYTES = 6
+const DISKPART_DELAY = 2000
+const DISKPART_RETRIES = 5
+
+/**
+ * @summary Generate a tmp filename with full path of OS' tmp dir
+ * @function
+ * @private
+ *
+ * @param {String} extension - temporary file extension
+ * @returns {String} filename
+ *
+ * @example
+ * const filename = tmpFilename('.sh');
+ */
+const tmpFilename = (extension) => {
+  const random = crypto.randomBytes(TMP_RANDOM_BYTES).toString('hex')
+  const filename = `etcher-diskpart-${random}${extension}`
```

Comment:
I wonder if we should use [`node-tmp`](https://github.com/raszi/node-tmp) instead of rolling our own unique file name generation code, given that I believe we have it on the dependency tree already.

Reply:
Remember that issue with temporary file spam and running out of file handles on drivelist (https://github.com/resin-io/etcher/issues/1571, https://github.com/resin-io-modules/drivelist/issues/190, https://github.com/resin-io-modules/drivelist/pull/191)? These couple of lines are more resilient than that module, and I'd rather not we use node-tmp anywhere :)

Reply:
Ah, good point. I'll approve then, but it'd be nice to get rid of the custom ESLint comments in various files first (by using a nested ESLint configuration).

Reply:
Yup, doing that in a minute or two :)